database/src/config.rs

```diff
@@ -16,151 +26,355 @@ use crate::{constants::DATABASE_FILENAME, resize::ResizeAlgorithm};
+    /// directory that contains all database files.
+    ///
+    /// If `None`, it will use the default directory [`cuprate_database_dir`].
+    pub fn low_power(db_directory: Option<PathBuf>) -> Self {
         let (db_directory, db_file) = Self::return_db_dir_and_file(db_directory);
         Self {
             db_directory,
             db_file,
-            sync_mode: SyncMode::Safe,
+            sync_mode: SyncMode::FastThenSafe,
             reader_threads: ReaderThreads::One,
             resize_algorithm: ResizeAlgorithm::new(),
         }
     }
 
+    /// Return the absolute [`Path`] to the database directory.
+    ///
+    /// This will be the `db_directory` given
+    /// (or default) during [`Config`] construction.
+    pub const fn db_directory(&self) -> &Cow<'_, Path> {
+        &self.db_directory
+    }
+
     /// Return the absolute [`Path`] to the database data file.
-    pub fn db_file_path(&self) -> &Path {
+    ///
+    /// This will be based off the `db_directory` given
+    /// (or default) during [`Config`] construction.
+    pub const fn db_file(&self) -> &Cow<'_, Path> {
         &self.db_file
     }
 }
 
 impl Default for Config {
+    /// Same as `Self::new(None)`.
+    ///
+    /// ```rust
+    /// # use cuprate_database::config::*;
+    /// assert_eq!(Config::default(), Config::new(None));
+    /// ```
     fn default() -> Self {
-        Self::new(None::<&'static Path>)
+        Self::new(None)
     }
 }
 
 //---------------------------------------------------------------------------------------------------- SyncMode
-/// TODO
-#[derive(Copy, Clone, Default, PartialEq, PartialOrd, Eq, Ord, Hash)]
+/// Disk synchronization mode.
+///
+/// This controls how/when the database syncs its data to disk.
+///
+/// Regardless of the variant chosen, dropping [`Env`](crate::Env)
+/// will always cause it to fully sync to disk.
+///
+/// # Sync vs Async
+/// All invariants except [`SyncMode::Fast`] are `synchronous`,
+/// as in the database will wait until the sync is finished before continuing.
+///
+/// `SyncMode::Fast` is `asynchronous`, meaning it will _NOT_
+/// wait until the sync is done before continuing. It will immediately
+/// move onto the next operation even if the database/OS has not responded.
+#[derive(Copy, Clone, Debug, Default, PartialEq, PartialOrd, Eq, Ord, Hash)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 #[cfg_attr(
     feature = "borsh",
     derive(borsh::BorshSerialize, borsh::BorshDeserialize)
 )]
 pub enum SyncMode {
-    /// Fully sync to disk per transaction.
+    /// Use [`SyncMode::Fast`] until fully synced,
+    /// then use [`SyncMode::Safe`].
+    ///
+    /// TODO: We could not bother with this and just implement
+    /// batching, which was the solution to slow syncs with `Safe`.
+    /// ref: <https://github.com/monero-project/monero/issues/1463>
+    /// solution: <https://github.com/monero-project/monero/pull/1506>
```

Comment:
Just noting that we will be batching blocks together but I still think this would be a good option to have.

Reply:
There are some open questions on how to implement this: https://github.com/Cuprate/cuprate/issues/78